tests/test_cli.py

```diff
@@ -0,0 +1,145 @@
+"""Unit tests for cli"""
+import json
+import sys
+import unittest
+import unittest.mock as mock
+from datetime import datetime
+
+import geospaas_processing.cli.download as cli_download
+import geospaas_processing.downloaders as downloaders
+from dateutil.tz import tzutc
+from django.contrib.gis.geos import GEOSGeometry
+from freezegun import freeze_time
+
+
+class DownlaodingCLITestCase(unittest.TestCase):
+    """Tests for the cli of downloading """
+
+    def setUp(self):
+        sys.argv = [
+            "",
+            '-d', "/test_folder/%Y_nh_polstere",
+            '-b', "200",
+            '-e', "2020-08-22",
+            '-r',
+            '-n',"100",
+            '-p',
+            '-g', "POLYGON ((-22 84, -22 74, 32 74, 32 84, -22 84))",
+            '-c', "/config_folder/config_file.yml",
+            '-q',
+            '{"dataseturi__uri__contains": "osisaf", "source__instrument__short_name__icontains": "AMSR2"}',
+            ]
+
+    def test_extract_arg(self):
+        """shall return the correct argument values based on the 'sys.argv' """
+        arg = cli_download.parse_args()
+        self.assertEqual(arg.begin, '200')
+        self.assertEqual(arg.config_file, '/config_folder/config_file.yml')
+        self.assertEqual(arg.down_dir, '/test_folder/%Y_nh_polstere')
+        self.assertEqual(arg.end, '2020-08-22')
+        self.assertEqual(arg.geometry, 'POLYGON ((-22 84, -22 74, 32 74, 32 84, -22 84))')
+        self.assertEqual(arg.number_per_day, '100')
+        self.assertEqual(
+            arg.query,
+            '{"dataseturi__uri__contains": "osisaf", "source__instrument__short_name__icontains": "AMSR2"}')
+        # testing the flag performance
```

Comment:
This is not performance testing.

Reply:
Yeah, I know that you mean by `performance` in programming.
I will change it with `# testing the flag enumeration`